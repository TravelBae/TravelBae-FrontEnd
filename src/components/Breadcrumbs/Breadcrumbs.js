import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import styles from './styles.scoped.css';

// Fungsi ini untuk membuat komponen Breadcrumbs
// parameter yang ada didalamnya merupakan props yang bisa diturunkan kedalam komponen ini
export default function Breadcrumbs({ className, items }) {
  const classes = [styles.root, className].filter(Boolean).join(' ');

  return (
    <nav className="root">
      {items.reduce((acc, curr, idx) => {
        if ((idx + 1) === items.length) {
          return [
            ...acc,
            idx ? (<Fragment key={`s${idx}`}>&nbsp;</Fragment>) : '',
            <p key={idx}>{curr.name}</p>];
        }
        return [
          ...acc,
          <Link key={idx} to={curr.url}>{curr.name}&nbsp;</Link>,
          <Fragment key={`s${idx}`}>&nbsp;/&nbsp;</Fragment>
        ];
      }, [])}
    </nav>
  );
}

Breadcrumbs.defaultProps = {
  className: '',
  items: [],
};

Breadcrumbs.propTypes = {
  className: PropTypes.string,
  items: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    url: PropTypes.string
  })),
};
