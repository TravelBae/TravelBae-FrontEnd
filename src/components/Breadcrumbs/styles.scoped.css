.root {
  color: #000000;
  display: flex;
}

.root > a {
  color: #3C82F6;
}

.root > a:hover {
  color: #92b7f6;
}
