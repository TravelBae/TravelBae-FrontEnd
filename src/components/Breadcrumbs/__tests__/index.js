import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Breadcrumbs from '../Breadcrumbs';

describe('src/components/elements/Breadcrumbs', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Breadcrumbs />);
    expect(tree).toMatchSnapshot();
  });

  test('with items', () => {
    const items = [
      { name: 'a', url: '/a' },
      { name: 'b', url: '/b' }
    ];
    const result = Breadcrumbs({ items });
    expect(result.props.children[2].props.children[1]).toBe();
  });

  test('items with 1 data', () => {
    const items = [
      { name: 'a', url: '/a' }
    ];
    const result = Breadcrumbs({ items });
    expect(result.props.children[0]).toBe('');
  });
});
