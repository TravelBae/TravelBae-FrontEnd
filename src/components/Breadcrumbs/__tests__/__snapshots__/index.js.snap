// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/Breadcrumbs render 1`] = `
<nav
  className=""
/>
`;
